tstest/clock.go

```diff
@@ -4,57 +4,495 @@
 package tstest
 
 import (
+	"container/heap"
 	"sync"
 	"time"
+
+	"tailscale.com/tstime"
+	"tailscale.com/util/mak"
 )
 
+// ClockBuilder is used to configure the initial settings for a Clock. Once
+// the settings are configured as desired, call ClockBuilder.Finish to get the
+// resulting Clock. ClockBuilder can only be used to create a single Clock.
+type ClockBuilder struct {
```

Comment:
the Java Builder pattern isn't commonly seen in Go. You trying to avoid an Options struct or the Functional Options Pattern?

Reply:
For some reason I forgot about the Options struct, which works fine here.